R/html_dependency.R

```diff
@@ -160,28 +160,54 @@ htmlDependencies <- function(x) {
 #' @rdname htmlDependencies
 #' @export
 `htmlDependencies<-` <- function(x, value) {
-  if (inherits(value, "html_dependency"))
-    value <- list(value)
-  attr(x, "html_dependencies") <- value
+  attr(x, "html_dependencies") <- asDependencies(value)
   x
 }
 
 #' @rdname htmlDependencies
 #' @export
 attachDependencies <- function(x, value, append = FALSE) {
-  if (append) {
-    if (inherits(value, "html_dependency"))
-      value <- list(value)
+  value <- asDependencies(value)
 
+  if (append) {
     old <- attr(x, "html_dependencies", TRUE)
     htmlDependencies(x) <- c(old, value)
-
   } else {
     htmlDependencies(x) <- value
   }
+
   return(x)
 }
 
+
+asDependencies <- function(x) {
```

Comment:
It would help to have a comment saying that this will _not_ execute `tagFunction `s and convert them into static dependencies.